### PR TITLE
Feature/inference opt sj

### PR DESCRIPTION
**Pull request by sjayanth21**
Added pre-processing (normalising and colorspace conversion only) as an option in the export script

### CODE REVIEW

File: models/experimental.py [hunk 1]
```diff
@@ -86,6 +88,39 @@ def forward(self, x, augment=False, profile=False, visualize=False):
...
+        Note: If using resize inside model, make sure the imgsz variable in export.py is changed accordingly to input image size
+        '''
+        transformed_tensor = input_tensor/255
+        transformed_tensor = torch.flip(transformed_tensor, [1])
+        return transformed_tensor
+
+# class which creates model with inbuilt preprocessing (defined in Custom_Layer)
+class Custom_Model(nn.Module):
+    def __init__(self, pretrained_model, nc, names, stride):
+        super(Custom_Model, self).__init__()
```
Comment (dkanagaraj) [inline models/experimental.py]: You don't have to send the arguments nc, names and stride. You can access them from pretrained_model itself

pretrained_model.nc
pretrained_model.names
pretrained_model.stride

File: models/experimental.py [hunk 2]
```diff
@@ -111,10 +146,18 @@ def attempt_load(weights, map_location=None, inplace=True, fuse=True):
             m._non_persistent_buffers_set = set()  # pytorch 1.6.0 compatibility
 
     if len(model) == 1:
-        return model[-1]  # return model
+        print("returned plain model")
+        print("model.stride is: ", model.stride)
+        return model[-1], model.stride   # return model
```
Comment (dkanagaraj) [inline models/experimental.py]: You don't have to return this also

File: models/experimental.py [hunk 3]
```diff
@@ -86,6 +88,39 @@ def forward(self, x, augment=False, profile=False, visualize=False):
         y = torch.cat(y, 1)  # nms ensemble
         return y, None  # inference, train output
 
+class Custom_Layer(nn.Module):
```
Comment (dkanagaraj) [inline models/experimental.py]: Keep this class at the end of the file so that conflicts are avoided while pulling their changes

File: models/experimental.py [hunk 4]
```diff
@@ -86,6 +88,39 @@ def forward(self, x, augment=False, profile=False, visualize=False):
...
+        interpolation = T.InterpolationMode.NEAREST
+        transformer = torch.nn.Sequential(T.Pad((0,64)),T.Resize((640,640),interpolation=interpolation), antialias=False)
+        transformed_tensor = transformer(input_tensor)
+        Note: If using resize inside model, make sure the imgsz variable in export.py is changed accordingly to input image size
+        '''
+        transformed_tensor = input_tensor/255
+        transformed_tensor = torch.flip(transformed_tensor, [1])
+        return transformed_tensor
+
+# class which creates model with inbuilt preprocessing (defined in Custom_Layer)
```
Comment (dkanagaraj) [inline models/experimental.py]: Add this as a doc string inside the class

File: models/experimental.py [hunk 5]
```diff
@@ -86,6 +88,39 @@ def forward(self, x, augment=False, profile=False, visualize=False):
...
+class Custom_Model(nn.Module):
```
Comment (dkanagaraj) [inline models/experimental.py]: CustomModel is recommended. No underscore in class name

**Review by dkanagaraj — CHANGES_REQUESTED** [review]

**Review by dkanagaraj — APPROVED** [review]
